src/System.Private.ServiceModel/tests/Common/Infrastructure/ServiceUtilHelper.cs

```diff
@@ -211,6 +217,60 @@ public static void EnsureClientCertificateInstalled()
     }
 
     // Tries to ensure that the peer trust certificate is installed into
+    // a local OSX keychain.  It obtains the certificate from the service
+    // utility endpoint and either installs it or verifies that a matching
+    // one is already installed.  InvalidOperationException will be thrown
+    // if an error occurred attempting to install the certificate. This
+    // method may be called multiple times but will attempt the installation
+    // once only.
+    public static void EnsureOSXKeychainCertificateInstalled()
+    {
+        if (!s_osXPeerCertAvailabilityChecked)
+        {
+            lock (s_certLock)
+            {
+                if (!s_osXPeerCertAvailabilityChecked)
+                {
+                    X509Certificate2 peerCertificate = null;
+                    string thumbprint = null;
+
+                    try
+                    {
+                        // Once only, we interrogate the service utility endpoint
+                        // for the server certificate and install it locally if it
+                        // is not already in the store.
+                        peerCertificate = InstallOSXPeerCertificateFromServer();
+
+                        // If we had a certificate from the service endpoint, verify it was installed
+                        // by retrieving it from the store by thumbprint.
+                        if (peerCertificate != null)
+                        {
+                            thumbprint = peerCertificate.Thumbprint;
+                            peerCertificate = CertificateManager.OSXLocalKeychainCertificateFromThumprint(thumbprint, validOnly: false);
+                            if (peerCertificate == null)
+                            {
+                                s_clientCertInstallErrorMessage =
+                                    $"Failed to find a server certificate matching thumbprint '{thumbprint}'";
+                            }
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        s_osXKeychainCertInstallErrorMessage = ex.ToString();
+                    }
+
+                    s_peerCertificate = peerCertificate;
+                    s_osXPeerCertAvailabilityChecked = true;
+                }
+            }
+        }
+
+        // If the installation failed, throw an exception everytime
+        // this method is called.
+        ThrowIfOSXKeychainCertificateInstallationError();
```

Comment:
The method is used here only. I don't think we need a separate method for throwing the error. That way `s_osXKeychainCertInstallErrorMessage` need not to be a static member.